.github/workflows/workflow-dispatch-job.yml

```diff
@@ -53,63 +53,127 @@ jobs:
         include: ${{ fromJSON(startsWith(inputs.runner, 'linux') && inputs.dummy_matrix || '[]') }}
     runs-on: ${{inputs.runner}}
     container:
-      options: -u root
-      image: ${{inputs.image}}
+      image: cruizba/ubuntu-dind:jammy-26.1.3
```

Comment:
This job now uses a docker-outside-of-docker (DOOD) strategy.

The GitHub Actions runner application mounts the host's docker socket `/var/run/docker.sock` into the container, so by using a container with the `docker` CLI, this container can launch docker containers using the host's docker daemon.

This allows us to run actions that require node20 (like `actions/checkout@v4` and `aws-actions/configure-aws-credentials@v4`) in the `cruizba/ubuntu-dind:jammy-26.1.3` container, and then launch our Ubuntu18.04-based GCC 6/7 containers to build and test CCCL.

There main inconvenience to this approach is that any container mounts have to match the paths of the runner host, not the paths as seen in the intermediate (`cruizba/ubuntu-dind`) container.

Note: I am using `cruizba/ubuntu-dind:jammy-26.1.3` instead of `docker:latest`, because GitHub doesn't support JS actions in alpine aarch64 containers:
```
Error: JavaScript Actions in Alpine containers are only supported on x64 Linux runners. Detected Linux Arm64
```

Reply:
Please add a comment here with these details.